src/nvs.rs

```diff
@@ -736,6 +805,80 @@ impl RawHandle for EspNvs<NvsEncrypted> {
         self.1
     }
 }
+impl RawHandle for EspNvs<NvsDefault> {
+    type Handle = nvs_handle_t;
+
+    fn handle(&self) -> Self::Handle {
+        self.1
+    }
+}
+
+#[cfg(esp_idf_version_at_least_5_2_0)]
+pub struct EspNvsKeys<'a> {
+    // The EspNvs must not be dropped while the iterator is still in use,
+    // this reference ensures that.
+    _nvs: PhantomData<&'a ()>,
+    raw_iter: nvs_iterator_t,
+    is_exhausted: bool,
+    key_name_buffer: [u8; 16],
+}
+
+#[cfg(esp_idf_version_at_least_5_2_0)]
+impl<'a> EspNvsKeys<'a> {
+    /// Returns the next key in the NVS namespace and its data type.
+    ///
+    /// After the last key is returned, this function will return `None` on subsequent calls.
+    pub fn next_key(&mut self) -> Option<(&str, NvsDataType)> {
+        if self.is_exhausted || self.raw_iter.is_null() {
+            return None;
+        }
+
+        let mut info: nvs_entry_info_t = Default::default();
```

Comment:
Micro opt for memory: you can just promote `nvs_entry_info_t` as a member to `EspNvsKeys` instead of having a separate key buffer, but oh well. I'll merge as -is and then maybe fix myself. No big deal. :)